serialization/dotnet/json/src/JsonSerializationWriter.cs

```diff
@@ -276,10 +270,13 @@ private void WriteAnyValue<T>(string key, T value)
                     WriteDateTimeOffsetValue(key, dto);
                     break;
                 case IEnumerable<object> coll:
-                    WriteCollectionOfPrimitiveValues(key, coll); // should we support collections of parsables here too?
+                    WriteCollectionOfPrimitiveValues(key, coll);
                     break;
                 case object o:
-                    WriteNonParsableObjectValue(key, o); // should we support parsables here too?
+                    if(o is IParsable parseable)
```

Comment:
Not doing this will result to reflection errors due to the presence of the AdditionalData property in `IParsable` objects. Furthermore, there is no need to handle this in the collection case as this function will still be recalled at the individual item level and be resolved as an `IParsable`

Reply:
can't break the case into two and not have the if?  `case IParsable p:`....

Reply:
That is genuinely better! Resolved via https://github.com/microsoft/kiota/pull/529/commits/fd02ffd086434992b122cf12ae0aa9b441741815